### PR TITLE
[WIP] RegExp grammar with nearley

### PR DESCRIPTION
**Pull request by targos**
/cc @TimothyGu I'll probably need your help with this.

### CODE REVIEW

File: src/grammar/RegExp.ne [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+GroupName ->
+    "<" RegExpIdentifierName ">"
+
+RegExpIdentifierName ->
+    RegExpIdentifierStart 
+  | RegExpIdentifierName RegExpIdentifierPart 
+
+RegExpIdentifierStart ->
+    [a-zA-Z]# UnicodeIDStart
```
Comment (targos) [inline src/grammar/RegExp.ne]: I have no idea how to encode `UnicodeIDStart` (and `UnicodeIDContinue` on line 93)

**Comment by TimothyGu** [comment]
Hmm, the negative lookaheads are problematic. I don’t know if nearley actually supports those.

**Comment by TimothyGu** [comment]
Just going to put that here: https://github.com/acornjs/acorn/blob/master/acorn/src/regexp.js looks like a good parser, though doesn’t look like it supplies an AST.

**Comment by TimothyGu** [comment]
Let's transfer your branch to devsnek/engine262: #71.

**Pull request closed without merging**